substrate-archive-backend/Cargo.toml

```diff
@@ -0,0 +1,47 @@
+[package]
+name = "substrate-archive-backend"
+version = "0.1.0"
+authors = ["Parity Technologies <admin@parity.io>"]
+edition = "2018"
+
+[dependencies]
+# External
+parking_lot = "0.10"
+futures = { version = "0.3" }
+hashbrown = { version = "0.8", features = ["inline-more"] }
+arc-swap = "0.4.7"
+log = "0.4"
+xtra = { version = "0.5.0-beta.4", features = ["with-smol-0_3"] }
+
+
+# Parity
+codec = { package = "parity-scale-codec", version = "1.3", default-features = false, features = ["derive", "full"] }
+kvdb = "0.7"
+kvdb-rocksdb = "0.9"
+hash-db = "0.15"
+parity-util-mem = "0.7"
+
+
+# Substrate
+sp-database = { git = "https://github.com/paritytech/substrate", package="sp-database" , rev = "647ad15565d7c35ecf00b73b12cccad9858780b9" }
```

Comment:
I think all the substrate dependencies will be pointed at this rev (it's the latest substrate that polkadot is using)

I was actually just in the process of updating everything to polkadot v0.8.25 :P

Reply:
Also on rustfmt -- I think the .rustfmt.toml has been committed for a while already -- it's just a matter of setting up our editors to use it and also have it in the CI pipeline to fail if code isn't formatted correctly

Reply:
re-rustfmt: lovely, for 11 months! But without failing CI it's perhaps not as useful as it could be... ;)
Also, what do you think about having a common "tools team" rustfmt? Perhaps the same as `jsonrpsee` (which comes from `parity-common`):

```toml
hard_tabs = true
max_width = 120
use_small_heuristics = "Max"
edition = "2018"
```

As for `rev`, I still think we need a comment documenting why it's set to what it's set to.

Reply:
It's a bit irksome that all of `archive` has to depend on specific substrate versions. It would be good if at least some of these new crates could be agnostic to the substrate version, so that only the binary crates would need to be built against a specific version. I don't think that's possible right now (or ever), but this tight coupling is a bit of a code smell.